tests/compat.rs

```diff
@@ -354,110 +300,33 @@ fn podman_container_launcher(
         r#"
 set -e
 
-# prepare rpm macros
-
-cat > ~/.rpmmacros << EOF_RPMMACROS
-%_signature gpg
-%_gpg_path /root/.gnupg
-%_gpg_name Package Manager
-%_gpgbin /usr/bin/gpg2
-%__gpg_sign_cmd %{__gpg} \
-    --batch \
-    --verbose \
-    --no-armor \
-    --keyid-format long \
-    --pinentry-mode error \
-    --no-secmem-warning \
-    %{?_gpg_digest_algo:--digest-algo %{_gpg_digest_algo}} \
-    --local-user "%{_gpg_name}" \
-    --sign \
-    --detach-sign \
-    --output %{__signature_filename} \
-    %{__plaintext_filename}
-EOF_RPMMACROS
-
-cat ~/.rpmmacros
-
-### either
-
-#cat > gpgkeyspec <<EOF
-#     %echo Generating a basic OpenPGP key
-#     Key-Type: RSA
-#     Key-Length: 2048
-#     Subkey-Type: RSA
-#     Subkey-Length: 2048
-#     Name-Real: Package Manager
-#     Name-Comment: unprotected
-#     Name-Email: pmanager@example.com
-#     Expire-Date: 0
-#     %no-ask-passphrase
-#     %no-protection
-#     %commit
-#     %echo done
-#EOF
-#gpg --batch --generate-key gpgkeyspec  2>&1
-
-### or (which has a couple of advantages regarding reproducibility)
-
-export PK=/assets/public_key.asc
-export SK=/assets/secret_key.asc
-
-gpg --allow-secret-key-import --import "${SK}" 2>&1
-gpg --import "${PK}" 2>&1
+pushd assets/
```

Comment:
This is unfortunately not present by default on quite a few linux distributions.

Reply:
I wasn't under the impression that these tests were meant to be totally generic, given that they call `yum`, `dnf`, `rpm`, use podman, etc.  

Is it OK, given that pretty much every distro we would care about (SUSE, Fedora, CentOS/RHEL) has it available?

Reply:
It adds a barrier, `rpm` can be reasonably expected, I i.e. don't have these installed on any of my machines (fedora 39).

Reply:
`pushd` and `popd` are bash builtins, they aren't installed separately, but they're available on any distro that uses bash as the shell, which includes Fedora.

```
dalley@thinkpad:~/Devel/rpm-dev/rpm-rs/tests/assets$ which pushd
/usr/bin/which: no pushd in (/home/dalley/.cargo/bin:/home/dalley/.local/bin:/home/dalley/bin:/usr/share/Modules/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin)
dalley@thinkpad:~/Devel/rpm-dev/rpm-rs/tests/assets$ pushd RPMS
~/Devel/rpm-dev/rpm-rs/tests/assets/RPMS ~/Devel/rpm-dev/rpm-rs/tests/assets
dalley@thinkpad:~/Devel/rpm-dev/rpm-rs/tests/assets/RPMS$ popd
~/Devel/rpm-dev/rpm-rs/tests/assets

```

Reply:
You are correct, as a `zsh` user I wasn't aware.